src/plugins/discover/public/application/components/sidebar/discover_field_details.tsx

```diff
@@ -39,6 +43,41 @@ export function DiscoverFieldDetails({
   onAddFilter,
 }: DiscoverFieldDetailsProps) {
   const warnings = getWarnings(field);
+  const [showVisualizeLink, setShowVisualizeLink] = useState<boolean>(false);
+  const [visualizeLink, setVisualizeLink] = useState<string | undefined>('');
+
+  useEffect(() => {
+    isFieldVisualizable(field, indexPattern.id, details.columns).then(
+      (flag) => {
+        setShowVisualizeLink(flag);
+        // get href only if Visualize button is enabled
+        getVisualizeHref(field, indexPattern.id, details.columns).then(
+          (uri) => {
+            setVisualizeLink(uri);
+          },
+          () => {
+            setVisualizeLink('');
+          }
+        );
+      },
+      () => {
+        setShowVisualizeLink(false);
+      }
+    );
+  }, [field, indexPattern.id, details.columns]);
+
+  const handleVisualizeLinkClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    if (
+      !event.defaultPrevented && // onClick prevented default
+      event.button === 0 && // ignore everything but left clicks
+      (!event.currentTarget.target || event.currentTarget.target === '_self') && // let browser handle "target=_blank" etc.
+      !(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey) // ignore clicks with modifier keys
+    ) {
+      // regular link click. let the uiActions code handle the navigation and show popup if needed
+      event.preventDefault();
+      triggerVisualizeActions(field, indexPattern.id, details.columns);
+    }
+  };
```

Comment:
A question about these checks, are they all necessary? since we're using `EuiLink` I'd assume that we know the target of the link

Reply:
and clicks with alt/ctrl/shift are not ignored, at least when I'm testing locally

Reply:
I will check it again, arch team provided this function to me but I suppose can be much simpler.